libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -4807,13 +4847,14 @@ rangeSize a b | a < 0 || b < 0 = error ("Negative Int in call to rangeSize: " ++
 rangeSize a b = NumSpecInterval (Just a) (Just b)
 
 -- | The widest interval whose largest element is admitted by the original spec
+--   This is a Size, so if the largest element is 0, then (NumSpecInterval (Just 0) (Just 0))
 maxSpec :: BaseUniverse fn => Specification fn Integer -> Specification fn Integer
 maxSpec TrueSpec = TrueSpec
 maxSpec s@(SuspendedSpec _ _) =
   constrained $ \x -> unsafeExists $ \y -> [y `satisfies` s, Assert ["maxSpec on SuspendedSpec"] (x <=. y)]
 maxSpec (ErrorSpec xs) = ErrorSpec xs
 maxSpec (MemberSpec []) = ErrorSpec ["empty MemberSec in maxSpec."]
-maxSpec (MemberSpec xs) = leqSpec (maximum xs)
+maxSpec (MemberSpec xs) = leqSpec (maximum (0 : xs)) -- TODO IS this right
```

Comment:
This is not right in general, it makes it impossible to use this function when we aren't talking about sizes.

Reply:
This is only used when computing cardianlities. So it is always a Size.  I have renamed it to maxSpecSize, and added a comment about how it should be used.